### ISSUE TITLE
[BUG] populate-daily-cache action masks errors

### ISSUE THREAD

**Issue by gerashegalov** [NVIDIA/spark-rapids#11796]
**Describe the bug**

The log for GH actions contains warnings and most importantly errors and yet succeeds

```
++ xargs -n 1 -I '{}' bash -c 'mvn -Ddist.jar.compress=false -DskipTests -Dmaven.scaladoc.skip -Dmaven.artifact.threads=10 --batch-mode -Dmaven.wagon.http.retryHandler.count=3 -Dmaven.wagon.httpconnectionManager.ttlSeconds=30 -Daether.connector.http.connectionMaxTtl=30 -Drapids.secondaryCacheDir=$HOME/.m2/repository/.sbt/1.0/zinc/org.scala-sbt --file pom.xml -Dbuildver={} de.qaware.maven:go-offline-maven-plugin:resolve-dependencies'
xargs: warning: options --max-args and --replace/-I/-i are mutually exclusive, ignoring previous --max-args value
Traceback (most recent call last):
  File "/home/runner/work/spark-rapids/spark-rapids/build/get_buildvers.py", line 73, in <module>
    print(_get_buildvers(sys.argv[1], sys.argv[2]))
  File "/home/runner/work/spark-rapids/spark-rapids/build/get_buildvers.py", line 65, in _get_buildvers
    return release_dict[buildvers]
KeyError: 'no_snapshots.buildvers'
```

We should fix both error masking and the underlying issues

**Steps/Code to reproduce bug**
Open a PR that triggers a new cache 

**Expected behavior**
Non-transient errors should fail the job.

**Environment details (please complete the following information)**
- GH hosted runners 

**Additional context**
N/A

**Comment by pxLi** [comment]
cc @YanxuanLiu to help, thanks

**Comment by gerashegalov** [comment]
Fixed by #11811
